src/main/java/bio/terra/service/dataset/flight/upgrade/predictableFileIds/ConvertToPredictableFileIdsBqCreateStageTableStep.java

```diff
@@ -0,0 +1,64 @@
+package bio.terra.service.dataset.flight.upgrade.predictableFileIds;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryDatasetPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.Map;
+import java.util.UUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ConvertToPredictableFileIdsBqCreateStageTableStep implements Step {
+  private static final Logger logger =
+      LoggerFactory.getLogger(ConvertToPredictableFileIdsBqCreateStageTableStep.class);
+
+  private final UUID datasetId;
+  private final DatasetService datasetService;
+  private final BigQueryDatasetPdao bigQueryDatasetPdao;
+
+  public ConvertToPredictableFileIdsBqCreateStageTableStep(
+      UUID datasetId, DatasetService datasetService, BigQueryDatasetPdao bigQueryDatasetPdao) {
+    this.datasetId = datasetId;
+    this.datasetService = datasetService;
+    this.bigQueryDatasetPdao = bigQueryDatasetPdao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    Dataset dataset = datasetService.retrieve(datasetId);
+
+    Map<UUID, UUID> oldToNewMappings =
+        ConvertFileIdUtils.readFlightMappings(context.getWorkingMap());
+
+    if (oldToNewMappings.isEmpty()) {
+      logger.info("No file ids to migrate");
+      return StepResult.getStepResultSuccess();
+    }
```

Comment:
I think this check might be missing for some of the other steps (e.g. ConvertToPredictableFileIdsUpdateFirestoreCollectionsStep) I would suggest moving this check to the flight level so we could skip the steps altogether, but since this is just migration code, not really worth it. 

There are a few places where we don't even end up using the oldToNewMappings and only check if there are any files to migrate. We could add a separate flight key that is just a boolean indicating where the map is empty.

Reply:
Yeah, the problem is that we don't know that it's empty until we're in step so we can't read it from the flight.  I only did the check where there the map being empty only causes a single thing to run or not (e.g. in `ConvertToPredictableFileIdsUpdateFirestoreCollectionsStep` we don't need the check...it just iterates on an empty collection)